nsiqcppstyle_exe.py

```diff
@@ -47,123 +46,168 @@
     console.Err.Error(msg)
     if usageOutput:
-        Usage()
+        get_parser().print_usage()
+        sys.exit(0)
     sys.exit(-1)
 
 
-def Usage():
-    print(
-        """
-======================================================================================
-Usage: nsiqcppstyle [Options]
-           targetdirectory
+def get_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=title
+        + """Apply custom rules on C/C++ code to report violations of your coding standards
 
 [Example]
    nsiqcppstyle .
    nsiqcppstyle targetdir
    nsiqcppstyle -f filefilterpath targetfilepath
+""",
+        epilog="""
+
+* If options like `--filter-string` and `--var` (which accept one or more arguments) are the last option, remember to separate the arguments and targets by adding a `--` before the targets. Eg:
+  --var key1:value key2:value -- my_target_path
 
-[Options]
-  -h            Show this help
-  -v            Show detail ouput(verbose mode)
-  -r            Show rule list
-  -o path       Set the output path. It's only applied when the output is csv or xml.
-  -f path       Set the filefilter path. If not provided, it uses the default filterpath
-                (target/filefilter.txt)
-                If you provide the file path (not a folder path) for the target,
-                -f option should be provided.
-  --filter-string=<filter string>
-                A single, valid filter file line.  This option may be repeated multiple
-                times.  Enables specifying the contents of a filter file without needing
-                to create one (e.g., in a read-only file system)
-  --var=key: value,key: value
-                provide the variables to customize the rule behavior.
-  --list-rules / -r  Show all rules available.
-                Add file extensions to be counted as assigned languages.
-  -s            Assign Filter scope name to be applied in this analysis
-  --output=     output format 'emacs', 'vs7', 'csv', 'xml' and 'eclipse'. Default value is vs7
-                emacs, vs7, eclipse output the result on the stdout in the form
-                that each tool recognizes.
-                csv and xml outputs the result on the file "nsiqcppstyle_result.csv"
-                "nsiqcppstyle_result.xml" respectively, if you don't provide -o option.
-  --ci          Continuous Integration mode. If this mode is on, this tool only reports summary.
-  --quiet / -q  Quiet mode. If this mode is on, this tool only reports errors.
-
-* nsiqcppstyle reports coding standard violations on C/C++ source code.
-* In default, it doesn't apply any rules on the source. If you want to apply rule,
-  they should be provided in the filefilter.txt file in following form.
+* By default, it doesn't apply any rules on the source. If you want to apply rule, they should be provided in the 'filefilter.txt' file in following format:
```

Comment:
Now I see all your changes :thumbsup:

Change "in following format" to "with the following format"

Reply:
```suggestion
* By default, it doesn't apply any rules on the source. If you want to apply rule, they should be provided in the 'filefilter.txt' file with the following format:
```